pxr/usd/usdUtils/testenv/testUsdUtilsValidators.cpp

```diff
@@ -0,0 +1,94 @@
+#include "pxr/usd/usd/validator.h"
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usdUtils/validatorTokens.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/base/arch/systemInfo.h"
+
+#include <iostream>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+static
+void
+TestUsdUsdzValidators()
+{
+    // This should be updated with every new validator added with
+    // UsdUsdzValidators keyword.
+    UsdValidationRegistry& registry = UsdValidationRegistry::GetInstance();
+    UsdValidatorMetadataVector metadata =
+            registry.GetValidatorMetadataForKeyword(
+                    UsdUtilsValidatorKeywordTokens->UsdUtilsValidators);
+    // Since other validators can be registered with a UsdUsdzValidators
+    // keyword, our validators registered in usd are a subset of the entire
+    // set.
+    std::set<TfToken> validatorMetadataNameSet;
+    for (const UsdValidatorMetadata &metadata : metadata) {
+        validatorMetadataNameSet.insert(metadata.name);
+    }
+
+    const std::set<TfToken> expectedValidatorNames =
+            {UsdUtilsValidatorNameTokens->packageEncapsulationValidator};
+
+    TF_AXIOM(std::includes(validatorMetadataNameSet.begin(),
+                           validatorMetadataNameSet.end(),
+                           expectedValidatorNames.begin(),
+                           expectedValidatorNames.end()));
+}
+
+static
+void
+TestPackageEncapsulationValidator()
+{
+    UsdValidationRegistry& registry = UsdValidationRegistry::GetInstance();
+
+    // Verify the validator exists
+    const UsdValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdUtilsValidatorNameTokens->packageEncapsulationValidator);
+
+    TF_AXIOM(validator);
+
+    const std::string testAssetDirectory = ArchGetCwd() + "/ctest/testUsdUtilsPackageEncapsulation/";
+
+    // Load the pre-created usdz stage with fully qualified paths to both a reference and an asset
+    const UsdStageRefPtr& stage = UsdStage::Open(testAssetDirectory + "fail.usdz");
```

Comment:
@tallytalwar I'm looking for feedback on this section. I'd like to do this in a way that does not depend on so many test files. I tried using [UsdUtilsCreateNewUsdzPackage](https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/usdUtils/usdzPackage.cpp#L81) however when using that on my `usda` file with fully qualified paths, the function corrects the paths internally so I don't get the failure condition I want. 

My next idea was to manually set the paths in code:

1. Create `subLayer.usda` (which is empty)
2. Create `image.jpg` (or some other asset file that's mostly empty)
3. Create `root.usda` that references the fully qualified paths of `subLayer.usda` & `image.jpg`
4. "Manually" turn `root.usda` into `test.usdz` (not sure what this function is, but not the above function since that corrects my paths, `usdzip` seems to work for this so maybe I can do the equivalent of that)
5. Verify errors
6. Update reference and asset paths in root.usda to be relative paths
7. "Manually" turn `root.usda` into `test.usdz` again (I guess here I could run `UsdUtilsCreateNewUsdzPackage` however to me it would be more clear to create the usdz the same way for someone reading the function
8. Verify the errors are gone
9. Delete `subLayer.usda`, `image.jpg`, 'root.usda', and `test.usdz`

This path seems to require no tests assets which is better but just wanted to get your thoughts before going forward incase you like the committed solution better or had some other idea